### PR TITLE
Refine code

### PR DESCRIPTION
**Pull request by blindpirate**
This PR introduces two changes:

1. Supports -DpublishStrategy=publishOnFailure.
2. Remove spaces in custom values so they can be searched more easily.

### CODE REVIEW

File: src/main/java/com/gradle/enterprise/conventions/customvalueprovider/ScanCustomValueNames.java [hunk 1]
```diff
@@ -1,10 +1,10 @@
 package com.gradle.enterprise.conventions.customvalueprovider;
 
 public interface ScanCustomValueNames {
-    String GIT_COMMIT_NAME = "Git Commit ID";
-    String GIT_STATUS = "Git Status";
-    String GIT_BRANCH_NAME = "Git Branch Name";
-    String BUILD_ID = "Build ID";
-    String IDEA_VERSION = "IDEA version";
+    String GIT_COMMIT_NAME = "Git_Commit_ID";
```
Comment (wolfs) [inline src/main/java/com/gradle/enterprise/conventions/customvalueprovider/ScanCustomValueNames.java:4]: I am not sure the `_` is better than the ` `. I think we should either use a human readable description or some case convention. Camel case or snake case are both fine. For example we could use `gitCommitId` or `git_commit_id`. Mixing case within the parts of the name doesn't seem to be a good idea for entering those keys in the search box.

Comment (blindpirate) [reply src/main/java/com/gradle/enterprise/conventions/customvalueprovider/ScanCustomValueNames.java:4]: given that we already have `watchFs`, let's stick in camel case.

Comment (wolfs) [reply src/main/java/com/gradle/enterprise/conventions/customvalueprovider/ScanCustomValueNames.java:4]: Camel case sounds good to me.

Comment (blindpirate) [reply src/main/java/com/gradle/enterprise/conventions/customvalueprovider/ScanCustomValueNames.java]: Done, PTAL!

File: src/test/java/com/gradle/enterprise/conventions/customvalueprovider/CICustomValueProviderIntegrationTest.java [hunk 2]
```diff
@@ -33,7 +33,7 @@ public void workWithJenkins() {
         succeeds("help");
 
         assertTrue(getConfiguredBuildScan().containsLink("Jenkins Build", "https://jenkins"));
-        assertTrue(getConfiguredBuildScan().containsValue("Build ID", "jenkins_id"));
+        assertTrue(getConfiguredBuildScan().containsValue("Build_ID", "jenkins_id"));
```
Comment (wolfs) [inline src/test/java/com/gradle/enterprise/conventions/customvalueprovider/CICustomValueProviderIntegrationTest.java:36]:
```suggestion
        assertTrue(getConfiguredBuildScan().containsValue("buildId", "jenkins_id"));
```

Shouldn't this also use the new value?

**Comment by blindpirate** [comment]
Thanks @wolfs !

**Pull request merged**